lib/core/models/settings.dart

```diff
@@ -0,0 +1,20 @@
+class Settings {
+  String? playerName;
+  bool enabledMusic;
+  bool enabledSound;
+
+  Settings({this.playerName, required this.enabledMusic, required this.enabledSound});
+
+  Settings.initial()
+      : playerName = null,
+        enabledMusic = true,
+        enabledSound = true;
+
+  Settings copyWith({String? playerName, bool? enabledMusic, bool? enabledSound}) {
```

Comment:
I don't know much about dart but copyWith sounds like a shitty name for a function that returns an object. try with settingsBuilder or settingsCreator

Reply:
That's such a beautiful name! I will never change it!